SAMtags.tex

```diff
@@ -234,13 +239,39 @@ \subsection{Barcodes}
 
 \begin{description}
 \item[BC:Z:\tagvalue{sequence}]
-Barcode sequence, with any quality scores stored in the {\tt QT} tag.
-
-\item[QT:Z:\tagvalue{qualities}]
-Phred quality of the barcode sequence in the {\tt BC} (or {\tt RT}) tag. Same encoding as {\sf QUAL}.
+Barcode sequence (Identifying the sample/library), with any quality scores (optionally) stored in the {\tt QT} tag.
+The {\tt BC} tag should match the {\tt QT} tag in length. 
+In the case of multiple unique molecular identifiers (e.g., one on each end of the template) the recommended implementation concatenates all the barcodes and qualities and places a dash (`{\tt -}') between the barcodes from the same template. 
+
+\item[BX:Z:\tagvalue{sequence}]
+Sequence bases from the unique molecular identifier. 
+These could be either corrected or uncorrected. Unlike MI, this may be non-unique in the file. 
+Should comprise of sequence base possibly separated by hyphens as {\tt RX}.
+In case that the bases represent the corrected bases, the original sequence can be stored in {\tt RX} (similar to OQ storing the original qualities of bases.)
+
+\item[MI:Z:\tagvalue{str}]
+Molecular Identifier. 
+A unique ID within the SAM file for the source molecule from which this read is derived. 
+All reads with the same MI tag represent the group of reads derived from the same source molecule. 
+
+\item[QT:Z:\tagvalue{qualities}] 
+Phred quality of the sample-barcode sequence in the {\tt BC} (or {\tt RT}) tag. 
+Same encoding as {\sf QUAL}, i.e., Phred score + 33.
+See {\tt BC} for more detail.
+
+\item[QX:Z:\tagvalue{qualities}] 
+Phred quality of the unique molecular identifier sequence in the {\tt RX} or {\tt BX} tag. 
+Same encoding as {\sf QUAL}, i.e., Phred score + 33.
+The {\tt RX} and {\tt BX} tags should match the {\tt QX} tag in length. 
+In the case of multiple unique molecular identifiers (e.g., one on each end of the template) the recommended implementation concatenates all the barcodes and qualities with no separator between the different barcodes.
```

Comment:
I don't think `BX` needs to be sequence bases: in the examples that I've seen it happens to be  the corrected sequence bases (of `RX`) with a `-1` suffix (where the 1 would be bumped to 2 for one library if you used two separately created libraries for the same sample in the same analysis).

I think `BX` is used to represent known uniqueness of the long molecule partitioning technology: reads from the same long molecule, will have the same partition, and so same `BX` identifier, but multiple long molecules may have the same `BX` identifier.

(Other information like alignments to a reference can be used to mark sets of reads which are very likely to have come from a particular molecule and such a set would be marked with `MI`: reads from a single molecule might have 0,1 or more `MI` values, but reads from two different long molecules should not have a common `MI`). 

[related comment above](#issuecomment-249290370)